documentation/ssr-and-hydration.mld

```diff
@@ -0,0 +1,291 @@
+  </p>
+%}
+
+This means that while rendering the application, there was a difference between the React tree pre-rendered from the server and the React tree rendered during the first render in the browser (hydration).
+
+As it's hard to see it just in words, let's go to an example with this simple code:
+
+{@reasonml[
+[@react.component]
+let make = () => {
+  let backgroundColor = switch%platform () {
+    | Server => "red"
+    | Client => "blue"
+  };
+
+  <div className=backgroundColor />;
+};
+]}
+
+To understand how [switch%platform] works, look at the {{!page-"browser_ppx"} browser_ppx page}
+And let's see the diagram above, which shows how the code will behave on each SSR step:
+
+{%html:<img src="./ssr-hydrate-issue.png" alt='Hydrate error' style="max-width: 100%; height: auto; vertical-align: middle;">%}
+
+* As we can see, there is a difference between III and IV, that will cause React to complain about content mismatch.
+
+This is because React understands that the markup from the server ([background-color: red](IV)) differs from the one it expects to be on the client at the hydration time ([background-color: blue](V)). 
+
+If we want different values on the server and client, we must do it after React hydrates the client or, commonly said, after *mount*.
+
+To work on it we must check if the client mounted and then change this value. This can be done with a hook that will be executed only after the client is hydrated.
+
+{@reasonml[
+module IsClientMounted = {
+  let use = () => {
+    let (isMounted, setIsMounted) = React.useState(() => false);
+
+    React.useEffect0(() => {
+      // This will be executed only on the client and after the hydration
+      setIsMounted(_ => true);
+      None;
+    });
+
+    isMounted;
+  };
+};
+
+[@react.component]
+let make = () => {
+  let isClientMounted = IsClientMounted.use();
+  let backgroundColor = isClientMounted ? "blue" : "red"
+  
+  <div className=backgroundColor />;
+};
+]}
+
+With a code like this one, the flow will be:
+
+{%html:<img src="./ssr-no-hydrate-issue.png" alt='Hydrate without issue' style="max-width: 100%; height: auto; vertical-align: middle;">%}
+
+** As we can see, there is no difference between III and IV, and React will be happy with it.
+
+In this case, we don't even need [switch%platform] to know if it's on client or native as [useEffects] only runs on client and the value is provided on the client and the content "blue" and "red" are native compatible.
+
+{1 Helpers that you can use}
+
+{2 IsClientMounted}
+
+IsClientMounted is a hook helper that you can create to check if the client is mounted.
+
+{@reasonml[
+let use = () => {
+  let (isMounted, setIsMounted) = React.useState(() => false);
+
+  React.useEffect0(() => {
+    setIsMounted(_ => true);
+    None;
+  });
+
+  isMounted;
+};
+]}
+
+{3 When using it:}
+
+You must use [IsClientMounted] whenever you have to deal with variables inside a Component body (Not the markup).
```

Comment:
I don't understand this sentence ^^

Reply:
The idea was to say that `IsClientMounted` is for variables, not for React Markup, that would be handle by `ServerOrClientRender`. But I agree that it's confusing, I updated to:

```
You should use [IsClientMounted] whenever you have to deal with variables inside a component. 
For React markup, take a look at {{!section-"server-or-client-render"} <ServerOrClientRender/>} 
or {{!section-"client-only"} <ClientOnly/>}.

Example:

{@reasonml[
let isFocusable =
  isClientMounted ? children->isFocusableElement : false;
]}
```
<img width="672" alt="image" src="https://github.com/user-attachments/assets/b04f1ef1-7e2a-4392-a8b9-7bc696158964" />